src/main/java/tictac7x/charges/items/U_ColossalPouch.java

```diff
@@ -0,0 +1,132 @@
+package tictac7x.charges.items;
+
+import static tictac7x.charges.store.ItemContainerId.INVENTORY;
+
+import com.google.gson.Gson;
+
+import net.runelite.api.Client;
+import net.runelite.api.ItemID;
+import net.runelite.api.Skill;
+import net.runelite.client.Notifier;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.chat.ChatMessageManager;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+import tictac7x.charges.ChargesImprovedConfig;
+import tictac7x.charges.item.ChargedItemWithStorage;
+import tictac7x.charges.item.storage.StorageItem;
+import tictac7x.charges.item.triggers.OnChatMessage;
+import tictac7x.charges.item.triggers.OnItemContainerChanged;
+import tictac7x.charges.item.triggers.OnStatChanged;
+import tictac7x.charges.item.triggers.TriggerBase;
+import tictac7x.charges.item.triggers.TriggerItem;
+import tictac7x.charges.store.ItemContainerId;
+import tictac7x.charges.store.Store;
+
+import static tictac7x.charges.ChargesImprovedPlugin.getNumberFromWordRepresentation;
+
+public class U_ColossalPouch extends ChargedItemWithStorage {
+    // TODO: save in config storage somehow
+    private int decayCount = 0;
+
+    public U_ColossalPouch(
+        final Client client,
+        final ClientThread client_thread,
+        final ConfigManager configs,
+        final ItemManager items,
+        final InfoBoxManager infoboxes,
+        final ChatMessageManager chat_messages,
+        final Notifier notifier,
+        final ChargesImprovedConfig config,
+        final Store store,
+        final Gson gson
+    ) {
+        super(ChargesImprovedConfig.colossal_pouch, ItemID.SMALL_POUCH, client, client_thread, configs, items, infoboxes, chat_messages, notifier, config, store, gson);
+        this.storage = storage
+            .storeableItems(
+                new StorageItem(ItemID.RUNE_ESSENCE).checkName("Rune essence"),
+                new StorageItem(ItemID.PURE_ESSENCE).checkName("Pure essence"),
+                new StorageItem(ItemID.DAEYALT_ESSENCE).checkName("Daeyalt essence"),
+                new StorageItem(ItemID.GUARDIAN_ESSENCE).checkName("Guardian essence")
+            )
+            .maximumTotalQuantity(40);
+
+        this.items = new TriggerItem[]{
+            new TriggerItem(ItemID.COLOSSAL_POUCH),
+            new TriggerItem(ItemID.COLOSSAL_POUCH_26786), // Degrated
+        };
+
+        this.triggers = new TriggerBase[]{
+            // Empty
+            new OnChatMessage("There is no essence in this pouch.").emptyStorage(),
+            new OnChatMessage("The rift becomes active!").emptyStorage(),
+
+            // Check
+            new OnChatMessage("There (?:is|are) (?<charges>.+?) (?<essence>pure|daeyalt|guardian)? ?essences? in this pouch.").matcherConsumer((m) -> {
```

Comment:
i can't believe they used words instead of numbers here, lol...

Reply:
Seriously, really weird choice.